### ISSUE TITLE
KG presentation improvements

### ISSUE THREAD

**Issue by jachro** [SwissDataScienceCenter/renku#591]
Area for improvements:
* the representation of a node, especially a command, has to be shorter;
* do not overwhelm a user with showing the whole graph;
* visualize different types of nodes;

Backend changes:
* A single node should contain info about the type, commit id, name/directory/command (name, description);
* The lineage query should return queried file with direct inputs and outputs;
* The direct inputs and outputs should contain links to the other level inputs/outputs;

UI changes:
* Change the format of the visualization for different node types;
* Adapt the query for the first piece of data;
* Add + button for leaves having other inputs/outputs;

**Comment by ciyer** [comment]
I like parts of this, but I think the cure is worse the the disease for some of them. 

The full graph for a project is the interesting thing about the graph. You want to see what data were used and how they were processed to produce which result in one picture.

Only showing direct ancestors and descendants will make it very tedious to interact with the KG. And *it will not solve the problem!* If the graph is unreadable, the graph will still be unreadable, it will just take 20 clicks to get to that point.

I think the better solution is to focus on making the full graph more readable. Shorter representations of nodes will help, as will grouping together files from a common directory.

Letting users make annotations to nodes will also help. Annotations may be text, or may be information like "lower importance" (i.e., make the node smaller), "default hidden", etc.

**Comment by rokroskar** [comment]
@ciyer I don't see the changes outlined at the top as being incompatible with what you propose. I think there can be a middle ground between showing the whole graph and showing just one level - perhaps neither extreme is the best solution. 

What is certainly missing from the description above is collapsing nodes that link to a container node, e.g. a directory or a dataset. This will already make the graph a lot easier to digest. 

In short, I understand the goals outlined above to be exactly about improving graph readability so maybe we just need to iterate on the wording so we're on the same page.

**Comment by jachro** [comment]
Hi,
I have to say, the more I think about the Graph presentation, the more doubts I have.  So I'm not entirely sure but maybe it would be good to take a step back and reflect where we are and where we want to go with that.
 
Personally, I think the whole issue is a bit problematic. As far as I remember, we agreed last Friday, that we don't really know what's really useful for users and that we are simply guessing on the next steps using our own views, not real users needs. To summarize, the only things we know are:
* we don't like what we have now :)
* we have projects with hundreds of commits and thousands of files we'd like to be able to cope with;
* we'd like to do something nice.

That's all good but do you think we are in a position that we can say the story is ready to play and we should be spending time working on it? Don't take me wrong, from a Dev point of view I'd love to take that work right now :) I know we decided to spend some time on that this sprint but should we maybe consider/keep in mind the following points:
* try to reach our users with what we have and ask them what they think (maybe don't show them the `vom_nat` project ;) Don't you think we should be using more our users to get some feedback and be more driven by that? The question is who can do that as we don't have a researcher?
* do the changes we agreed on and then get some feedback (accepting we'll spend a considerable amount of time doing them and then people can say we'd like it to be a bit or completely different);
* do changes and iterate over them and trust that devs know how it should be;
* look at our backlog to see if there are maybe more urgent features (even using our KG :) Should we talk to @cchoirat ?

Sorry for that probably quite subjective thoughts but I see we're all having quite strong opinions and I'm not sure if we are not getting too Dev biased.

**Comment by rokroskar** [comment]
I share your concerns. I see the graph display right now as a feature that is too broken to be evaluated. So I see this work as making it into a minimal usable state, which will allow us to actually get user feedback. I don't want to go crazy with what we implement, but we need to either make it presentable or get rid of it completely (also an option).

**Comment by jachro** [comment]
:+1

**Comment by rokroskar** [comment]
The reason I thought we could put this on the stack of things now is that some prep work is still needed before the proper work on integrating datasets (higher priority) can begin. So if we can clean the graph up in 1-2 weeks while the dataset integration is prepped it's a win-win.
